apps/upskii/src/app/[locale]/(dashboard)/[wsId]/certificate/[certificateId]/download-button-pdf.tsx

```diff
@@ -0,0 +1,75 @@
+'use client';
+
+import { BASE_URL } from '@/constants/common';
+import { Button } from '@tuturuuu/ui/button';
+import { FileText } from '@tuturuuu/ui/icons';
+import { useTranslations } from 'next-intl';
+import { useCallback } from 'react';
+
+export function DownloadButtonPDF({
+  certificateId,
+  wsId,
+  className = '',
+  variant = 'outline',
+}: {
+  certificateId: string;
+  wsId: string;
+  className?: string;
+  variant?: 'default' | 'outline' | 'ghost' | 'link' | 'destructive';
+}) {
+  const t = useTranslations('certificates');
+
+  const handleDownload = useCallback(async () => {
+    try {
+      const res = await fetch(
+        `${BASE_URL}/api/v1/certificates/${certificateId}/generate`,
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            certID: certificateId,
+            title: t('title'),
+            certifyText: t('certify_text'),
+            completionText: t('completion_text'),
+            offeredBy: t('offered_by'),
+            completionDateLabel: t('completion_date'),
+            certificateIdLabel: t('certificate_id'),
+            wsId: wsId,
+          }),
```

Comment:
Please optimize the behavior for this API. Do not pass text to the server just for rendering purposes.